addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/WakeOnLanUtility.java

```diff
@@ -0,0 +1,126 @@
+import java.net.InetAddress;
+import java.net.InterfaceAddress;
+import java.net.NetworkInterface;
+import java.util.Enumeration;
+
+import org.openhab.binding.samsungtv.internal.handler.SamsungTvHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class with utility functions to support Wake On Lan (WOL)
+ *
+ * @author Arjan Mels - Initial contribution
+ *
+ */
+public class WakeOnLanUtility {
+
+    private static Logger logger = LoggerFactory.getLogger(SamsungTvHandler.class);
+
+    /**
+     * Get MAC address for host
+     * uses "arping" tool
+     *
+     * @param hostName Host Name (or IP address) of host to retrieve MAC address for
+     * @return MAC address
+     */
+    public static String getMACAddress(String hostName) {
+        try {
+            Process proc = Runtime.getRuntime().exec("arping -r -c 1 -C 1 " + hostName);
+            proc.waitFor();
+            BufferedReader stdInput = new BufferedReader(new InputStreamReader(proc.getInputStream()));
+            String macAddress = stdInput.readLine();
+            logger.info("MAC address of host {} is {}", hostName, macAddress);
+            return macAddress;
+
+        } catch (Exception e) {
+            logger.info("Problem getting MAC address: {}", e.getMessage());
+        }
+        return null;
+    }
+
+    /**
+     * Send single WOL (Wake On Lan) package on all interfaces
+     *
+     * @macAddress MAC address to send WOL package to
+     */
+    public static void sendWOLPacket(String macAddress) {
+        byte[] bytes = getWOLPackage(macAddress);
+
+        try {
+            Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+            while (interfaces.hasMoreElements()) {
+                NetworkInterface networkInterface = interfaces.nextElement();
+                if (networkInterface.isLoopback()) {
+                    continue; // Do not want to use the loopback interface.
+                }
+                for (InterfaceAddress interfaceAddress : networkInterface.getInterfaceAddresses()) {
+                    InetAddress broadcast = interfaceAddress.getBroadcast();
+                    if (broadcast == null) {
+                        continue;
+                    }
+
+                    try {
+                        DatagramPacket packet = new DatagramPacket(bytes, bytes.length, broadcast, 9);
+                        DatagramSocket socket = new DatagramSocket();
+                        socket.send(packet);
+                        socket.close();
+                        logger.trace("Sent WOL packet to {} {}", broadcast, macAddress);
+                    } catch (Exception e) {
+                        logger.warn("Problem sending WOL packet to {} {}", broadcast, macAddress);
+                    }
+                }
+            }
+
+        } catch (Exception e) {
+            logger.warn("Problem with interface while sending WOL packet to {}", macAddress);
```

Comment:
this should not log info, only debug

Reply:
Why should I not log a warning if the WOL packet cannot be send? (The TV will not turn on). Is there a guideline somewhere on which log level to use for what?